### PR TITLE
[MM-59357] Migrate tooltips of "components/custom_status/custom_status_suggestion.tsx" to WithTooltip

### LINKED ISSUE #27582: Migrate tooltips of "components/custom_status/custom_status_suggestion.tsx" to WithTooltip
The purpose of this ticket is to eliminate the usages of OverlayTrigger of “components/overlay_trigger” and Tooltip of “components/tooltip” in the file `components/custom_status/custom_status_suggestion.tsx`.` These should be replaced with the WithTooltip component from "components/with_tooltip". For reference, please follow this PR [https://git](https://github.com/mattermost/mattermost/pull/27145)"[ub.com/mattermost/mattermost/pull/27145](https://github.com/mattermost/mattermost/pull/27145)

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-59357

### PR DESCRIPTION
**Pull request by ZubairImtiaz3**
#### Summary
The changes made ensure that the tooltips in components/custom_status/custom_status_suggestion.tsx now utilize the WithTooltip component, replacing the previous usage of OverlayTrigger and Tooltip.

#### Ticket Link
Fixes #27582
Jira: [MM-59357](https://mattermost.atlassian.net/browse/MM-59357)

### Screenshot
<a href="https://ibb.co/Qp46cgS">Screenshot-2024-07-18-001828</a>

#### Release Note

```release-note
NONE
```

[MM-59357]: https://mattermost.atlassian.net/browse/MM-59357?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webapp/channels/src/components/custom_status/custom_status_suggestion.tsx [hunk 1]
```diff
@@ -42,25 +41,19 @@ const CustomStatusSuggestion: React.FC<Props> = (props: Props) => {
...
-                delayShow={Constants.OVERLAY_TIME_DELAY}
+        <div className='suggestion-clear'>
+            <WithTooltip
                 placement='top'
-                overlay={
-                    <Tooltip id='clear-recent-custom-status'>
-                        {'Clear'}
-                    </Tooltip>
-                }
+                title='Clear'
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/custom_status/custom_status_suggestion.tsx:47]: Just a bit more of an ask : i noticed that we didnt translate this word, can you use format intl hook to add translation here?

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by Rajat-Dabade — APPROVED** [review]
Apart from @M-ZubairAhmed comment's everything else LGTM 👍

File: webapp/channels/src/components/custom_status/custom_status_suggestion.tsx (hunk 1 above)
Comment (ZubairImtiaz3) [reply webapp/channels/src/components/custom_status/custom_status_suggestion.tsx]: Sure, I've added the translation for 'Clear' using FormattedMessage.

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**